Documentation/ExtensionArchitecture/Tutorials/Tea/Model.rst

```diff
@@ -0,0 +1,298 @@
+backend input form in the order they should be displayed.
+
+..  hint::
+    It is possible to influence the display of the fields in more sophisticated
+    ways: You can order them in tabs, put them into palettes etc. See
+    :ref:`TCA reference, showitem <t3tca:types-properties-showitem>` for
+    details.
+
+Result - the complete TCA
+--------------------------
+
+Have a look at the complete file
+`EXT:tea/Configuration/TCA/tx_tea_domain_model_product_tea.php <https://github.com/TYPO3-Documentation/tea/blob/main/Configuration/TCA/tx_tea_domain_model_product_tea.php>`__.
+
+The edit form for tea records in the backend now looks like this:
+
+..  figure:: /Images/ManualScreenshots/ExtensionArchitecture/Tutorials/Tea/TeaEditForm.png
+    :class: with-shadow
+
+    The complete input form for a tea record
+
+The list of teas in the module :guilabel:`Web -> List` looks like this:
+
+
+..  figure:: /Images/ManualScreenshots/ExtensionArchitecture/Tutorials/Tea/TeaList.png
+    :class: with-shadow
+
+    A list of teas in the backend
+
+..  note::
+    Up to this point we have only used TYPO3 Core features. You can create
+    tables and backend forms exactly the same way without using Extbase.
+
+The Extbase model
+=================
+
+It is a common practice - though not mandatory - to use PHP class to store the
+data while working on it.
+
+The model is an additional abstraction from the database. It enables the
+developer to require variable types beyond what the database abstraction layer
+offers. The model can also be used to apply validators to the properties
+of the model and to specify relationship types (should relatives be loaded
+lazily? Should they be deleted if this object is deleted?).
+
+Extbase models extend the class
+:php:`TYPO3\CMS\Extbase\DomainObject\AbstractEntity` provided by Extbase.
+The parent classes of this class already offer methods needed for persistence in
+to database, accessing the identifier :php:`uid` etc.
+
+.. include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Model/TeaProperties.rst.txt
+
+..  attention::
+    All properties of the model have to have the modifier :php:`protected` or
+    :php:`public`. :php:`private` properties are not supported as properties have to
+    be accessed by the repository and persistence and layers internally.
+
+    If you want to prevent developers from extending you model, and though
+    accessing the properties of you model, you can make the class of the model
+    final.
+
+For all :php:`protected` properties we need at least a getter of the according
+name. If the property should be writable within the Extbase workflows it also
+has to have a getter. Properties that are only set in the backend forms do not
+need a setter.
+
+Example for the property :php:`title`:
+
+..  include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Model/TeaTitle.rst.txt
+
+The getter for the image also has to resolve the lazy loading:
```

Comment:
You haven't described lazy loading on this page. Maybe a good idea to add a description of it further up.